cirq/__init__.py

```diff
@@ -168,13 +168,25 @@
     moment_by_moment_schedule,
 )
 
+from cirq.sim import (
+    dirac_notation,
+    SampleSimulator,
```

Comment:
A major UX problem with having this class at the top level is that people exploring the API will go "Oh, SampleSimulator, that must be the simulator I use to get samples. I'll remember that.". Then it doesn't give them samples, because it's an interface, and they feel tricked.

The same goes for the other `XSimulator` base classes.

I'm not sure what our naming convention should be to avoid this. Perhaps `SupportsSampleSimulator` or `AbstractSampleSimulator` or `SampleSimulatorInterface` or something along those lines.

Reply:
Maybe we can move the simulate word.

SimulatesSamples
SimulatesFinalWaveFunction
SimulatesIntermediateWaveFunction

I'm not as worried about the length of these names, since they actually are only used by those implementing the interface, so not really part of core use.

Thoughts?